src/Parser/Druid/Feral/Modules/Traits/AshamanesRip.js

```diff
@@ -0,0 +1,76 @@
+import React from 'react';
+
+import Analyzer from 'Parser/Core/Analyzer';
+import Enemies from 'Parser/Core/Modules/Enemies';
+import Combatants from 'Parser/Core/Modules/Combatants';
+
+import SPELLS from 'common/SPELLS';
+import SpellIcon from 'common/SpellIcon';
+import SpellLink from 'common/SpellLink';
+import { formatNumber, formatPercentage } from 'common/format';
+import Wrapper from 'common/Wrapper';
+import SmallStatisticBox, { STATISTIC_ORDER } from 'Main/SmallStatisticBox';
+
+/**
+ * Ashamane's Bite (Artifact Trait)
+ * Combo point generators on targets with your Rip have a 10% chance to place a duplicate of that Rip on the target.
+ * There is a maximum of one Ashamane's Rip active on a target at once, and it will be overriden if it procs again. 
+ * Ashamane's Rip copies the damage and duration of the Rip at the moment it is created.
+ * For this reason it's beneficial to pool energy before refreshing Rip so you can use many combo point generators when the Rip still has a long duration in the hope of copying a long duration rip.
+ */
+class AshamanesRip extends Analyzer {
+  static dependencies = {
+    enemies: Enemies,
+    combatants: Combatants,
+  };
+
+  damage = 0;
+
+  on_initialized() {
+    this.active = this.combatants.selected.traitsBySpellId[SPELLS.ASHAMANES_BITE.id] === 1;
+  }
+
+  on_byPlayer_damage(event) {
+    const spellId = event.ability.guid;
+
+    if (spellId !== SPELLS.ASHAMANES_RIP.id){
+      return;
+    }
+    
+    this.damage += (event.amount || 0) + (event.absorbed || 0) + (event.overkill || 0);
+  }
+
+  get ashamanesRipUptime() {
+    return this.enemies.getBuffUptime(SPELLS.ASHAMANES_RIP.id) / this.owner.fightDuration;
+  }
+
+  suggestions(when) {
```

Comment:
It is prefered that you use the new format for suggestions, you can see an example here:
https://github.com/WoWAnalyzer/WoWAnalyzer/blob/master/src/Parser/Druid/Balance/Modules/Features/MoonfireUptime.js